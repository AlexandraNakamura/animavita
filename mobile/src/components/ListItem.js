import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet } from 'react-native';
import TouchableScale from 'react-native-touchable-scale';
import LinearGradient from 'react-native-linear-gradient';
import { ListItem as Item } from 'react-native-elements';

const styles = StyleSheet.create({
  title: { color: 'white', fontWeight: '500', fontSize: 15 },
  container: { borderRadius: 15, marginVertical: 5 },
});

const ListItem = ({
  title, subtitle, avatar, onPress,
}) => (
  <Item
    Component={TouchableScale}
    containerStyle={styles.container}
    friction={90}
    tension={100}
    activeScale={0.95}
    linearGradientProps={{
      colors: ['#0AC4BA', '#2BDA8E'],
      start: { x: 1, y: 0 },
      end: { x: 0.2, y: 0 },
    }}
    ViewComponent={LinearGradient}
    leftAvatar={{
      rounded: true,
      source: { uri: avatar },
    }}
    title={title}
    titleStyle={styles.title}
    onPress={onPress}
    chevronColor="white"
    chevron
  />
);

ListItem.propTypes = {
  title: PropTypes.string.isRequired,
  avatar: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
};

export default ListItem;
