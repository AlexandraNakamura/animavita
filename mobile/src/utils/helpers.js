import AsyncStorage from '@react-native-community/async-storage';

export const sanitizeUsername = (name) => {
  if (name && name.length > 8) {
    return `\n ${name}`;
  }

  return name;
};

export const isEmpty = array => array instanceof Array && array.length === 0;

export const getToken = async () => JSON.parse(await AsyncStorage.getItem('@animavita:authToken')) || {};
