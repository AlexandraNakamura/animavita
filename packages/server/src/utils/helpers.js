import AWS from 'aws-sdk';

require('dotenv').config();

const {
  S3_KEY, S3_SECRET, S3_BUCKET, S3_REGION
} = process.env;

export const getAnimalGender = (gender) => {
  const genders = [
    {
      key: 'male',
      translate: 'Macho'
    },
    {
      key: 'female',
      translate: 'Fêmea'
    }
  ];

  return genders.find(element => element.key === gender).translate;
};

export const getAnimalSize = (size) => {
  const sizes = [
    {
      key: 'small',
      translate: 'Pequeno'
    },
    {
      key: 'medium',
      translate: 'Médio'
    },
    {
      key: 'bigger',
      translate: 'Grande'
    }
  ];

  return sizes.find(element => element.key === size).translate;
};

export const uploadFile = async (upload) => {
  let url = '';
  // eslint-disable-next-line new-cap
  const base64Data = new Buffer.from(
    upload.preview.uri.replace(/^data:image\/\w+;base64,/, ''),
    'base64'
  );

  const params = {
    Bucket: S3_BUCKET,
    Key: upload.image.name,
    Body: base64Data,
    ACL: 'public-read',
    ContentEncoding: 'base64',
    ContentType: upload.image.type
  };

  const s3 = new AWS.S3({
    accessKeyId: S3_KEY,
    secretAccessKey: S3_SECRET,
    region: S3_REGION
  });

  try {
    const { Location } = await s3.upload(params).promise();
    url = Location;
  } catch (err) {
    throw err;
  }

  return url;
};
