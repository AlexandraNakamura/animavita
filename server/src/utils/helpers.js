export const getAnimalGender = (gender) => {
  const genders = [
    {
      key: 'male',
      trans: 'Macho'
    },
    {
      key: 'female',
      trans: 'Fêmea'
    }
  ];

  return genders.find(element => element.key === gender).trans;
};

export const getAnimalSize = (size) => {
  const sizes = [
    {
      key: 'small',
      trans: 'Pequeno'
    },
    {
      key: 'medium',
      trans: 'Médio'
    },
    {
      key: 'bigger',
      trans: 'Grande'
    }
  ];

  return sizes.find(element => element.key === size).trans;
};
