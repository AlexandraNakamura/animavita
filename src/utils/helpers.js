import AsyncStorage from '@react-native-community/async-storage';

export const sanitizeUsername = (name) => {
  if (name && name.length > 8) {
    return `\n ${name}`;
  }

  return name;
};

export const isEmpty = (array) => {
  if (array instanceof Array && array.length === 0) {
    return true;
  }

  return false;
};

export const getUser = async () => JSON.parse(await AsyncStorage.getItem('@animativa:user'));
