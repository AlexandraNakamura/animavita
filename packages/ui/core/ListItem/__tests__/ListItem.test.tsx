import React from 'react';
import {Text} from 'react-native';
import {fireEvent} from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import {Mount, ThemeContext} from '../../../tests/helpers';
import ListItem from '../index';

const defaultProps = {
  onPress: jest.fn().mockName('onPress'),
};

const mountFactory = propOverrides => {
  const {children} = propOverrides;
  const listItemProps = {...defaultProps, ...propOverrides};

  const element = Mount(<ListItem {...listItemProps}>{children}</ListItem>);

  return element;
};

describe('List Item', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('calls onPress', () => {
    const cb = jest.fn();
    const {getByText} = mountFactory({children: <Text>Press me!</Text>, onPress: cb});

    fireEvent.press(getByText('Press me!'));

    expect(cb).toHaveBeenCalled();
  });

  it('renders correctly', () => {
    const tree = renderer.create(ThemeContext(<ListItem gradient onPress={defaultProps.onPress} />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
