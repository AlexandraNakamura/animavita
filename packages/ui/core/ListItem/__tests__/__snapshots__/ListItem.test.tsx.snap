// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Item renders correctly 1`] = `
<View
  colors={
    Array [
      "#1BC0B5",
      "#2ED68F",
    ]
  }
  end={
    Object {
      "x": 1,
      "y": 1,
    }
  }
  gradient={true}
  locations={
    Array [
      0.1,
      0.9,
    ]
  }
  start={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  style={
    Array [
      Object {
        "borderRadius": 6,
      },
    ]
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    gradient={true}
    onPress={[MockFunction onPress]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 6,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 10,
          "paddingRight": 10,
          "paddingTop": 10,
        },
      ]
    }
  />
</View>
`;
